geps/gep-1897.md

```diff
@@ -120,12 +137,335 @@ was followed by intense discussion and closed in favor of a downsize in scope.
+If CACertRefs is unspecified, then CertRefs must be specified with at least one entry for a valid configuration.
+If CACertRefs is "system" and there are no system trusted certificates or the implementation doesn't define system
+trusted certificates, then a TLS connection must fail.
+
+The `Hostname` field is required and is to be used to configure the SNI the Gateway should use to connect to the backend.
+Implementations must validate that at least one name in the certificate served by the backend matches this field.
+We originally proposed using a list of allowed Subject Alternative Names, but determined that this was [not needed in
+the first round](https://github.com/kubernetes-sigs/gateway-api/pull/2113#issuecomment-1696127092),
+but may be added in the future.
+
+We originally proposed allowing the configuration of expected TLS versions, but determined that this was [not needed in
+the first round](https://github.com/kubernetes-sigs/gateway-api/pull/2113#issuecomment-1696127092).
+
+Thus, the following additions would be made to the Gateway API:
+
+```go
+import "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+// TLSConnectionPolicy provides a way to publish TLS configuration
+// that enables a gateway client to connect to a backend pod.
+type TLSConnectionPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of TLSConnectionPolicy.
+    Spec TLSConnectionPolicySpec `json:"spec"`
+
+    // Status defines the current state of TLSConnectionPolicy.
+    Status TLSConnectionPolicyStatus `json:"status,omitempty"`
+}
+
+// TLSConnectionPolicySpec defines the desired state of
+// TLSConnectionPolicy.
+// Note: there is no Override or Default policy configuration.
+//
+// Support: Core
+type TLSConnectionPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    // Note that this config applies to the entire referenced resource
+    // by default, but this default may change in the future to provide
+    // a more granular application of the policy.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // TLS contains TLS connection policy configuration.
+    TLS *TLSConnectionPolicyConfig `json:”tls”`
+}
+
+// TLSConnectionPolicyConfig contains TLS connection policy configuration.
+type TLSConnectionPolicyConfig struct {
+    // CertRefs contains one or more references to
+    // Kubernetes objects that contain PEM-encoded TLS certificates,
+    // which are used to establish a TLS handshake between the gateway
+    // and backend pod.
+    //
+    // If the CertRefs is empty, then the system trusted
+    // certificates should be used. If there are none, or the
+    // implementation doesn't define system trusted certificates,
+    // then a TLS connection must fail.
```

Comment:
We should remove this because this behavior is now handled by the other field (`CACertRefs` here, but as I said above, I think it needs a new name).

Reply:
I think a user could accidentally try to use System certs but the system doesn't have them, so I think it should stay in.